packages/blade/src/components/FileUpload/_decisions/decisions.md

```diff
@@ -0,0 +1,359 @@
+   */
+  name?: string;
+  /**
+   * Default list of files that have been uploaded, useful when the component is uncontrolled
+   */
+  defaultFileList?: BladeFileList;
+  /**
+   * List of files that have been selected/uploaded, useful when the component is controlled
+   */
+  fileList?: BladeFileList;
+  /**
+   * Limit the number of files that can be uploaded
+   */
+  maxCount?: number;
+  /**
+   * Limit the size of the uploaded files (in bytes)
+   */
+  maxSize?: number;
+  /**
+   * Callback function triggered when files are selected
+   */
+  onChange?: ({ name, fileList }: { name: string; fileList: BladeFileList }) => void;
+  /**
+   * Callback function triggered when the preview icon is clicked
+   */
+  onPreview?: (previewedFile: BladeFile) => void;
+  /**
+   * Callback function triggered when a file is removed
+   */
+  onRemove?: (removedFile: BladeFile) => void;
+  /**
+   * Callback function executed when files are dropped into the upload area
+   */
+  onDrop?: (event: React.DragEvent) => void;
+  /**
+   * State indicating whether there is an error in the FileUpload component
+   */
+  validationState?: 'none' | 'error';
+  /**
+   * Additional text providing assistance or guidance
+   */
+  helpText?: string;
+  /**
+   * Text indicating an error state
+   */
+  errorText?: string;
+};
+```
+
+## Usage
+
+Primarily, there are two types of usage for the `FileUpload` component:
+
+### Submit Button File Upload (Uncontrolled)
+
+https://github.com/razorpay/blade/assets/46647141/d78ec694-f6f7-413d-a32c-58da8f2582fb
+
+### Direct Upload with Progress Display (Controlled)
+
+https://github.com/razorpay/blade/assets/46647141/cb690625-011b-4c20-b610-9af5f0c5e077
+
+## Examples
+
+Here are a few illustrative examples showcasing the utilization of the `FileUpload` component:
+
+### Uncontrolled Usage
+
+#### Single File selection
+
+```jsx
+import React, { useRef } from 'react';
+import axios from 'axios';
+import { Box, FileUpload } from '@razorpay/blade/components';
+
+const UncontrolledSingleFileUploadForm = () => {
+  const fileUploadRef = useRef();
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+
+    // Create a FormData object to append files
+    const formData = new FormData();
+    formData.append('files', fileUploadRef.current.files[0]);
+    try {
+      // Simulate a file upload using axios
+      const response = await axios.post(
+        'https://run.mocky.io/v3/bb0b32f0-fc54-4d78-9c9b-08b3a4d8f7c5',
+        formData,
+        {
+          headers: {
+            'content-type': 'multipart/form-data',
+          },
+        },
+      );
+
+      // Handle success, reset form, etc.
+      console.log('Files uploaded successfully:', response.data);
+    } catch (error) {
+      // Handle errors
+      console.error('File upload failed:', error.message);
+    }
+  };
+
+  return (
+    <Box>
+      <form onSubmit={handleSubmit}>
+        <FileUpload
```

Comment:
This file list is always rendered below input? any usecase where it needs to be rendered at different place?
<img width="693" alt="image" src="https://github.com/razorpay/blade/assets/30949385/b7b0f743-6bce-49db-bb1d-91b1958ec01c">

Reply:
Not as of now, if we give users control over this we won't be able to standardize it.